sched/semaphore/sem_holder.c

```diff
@@ -86,30 +87,36 @@ static inline FAR struct semholder_s *nxsem_allocholder(sem_t *sem)
       g_freeholders    = pholder->flink;
       pholder->flink   = sem->hhead;
       sem->hhead       = pholder;
-
-      /* Make sure the initial count is zero */
-
-      pholder->counts  = 0;
     }
 #else
   if (sem->holder[0].htcb == NULL)
     {
       pholder          = &sem->holder[0];
-      pholder->counts  = 0;
     }
   else if (sem->holder[1].htcb == NULL)
     {
       pholder          = &sem->holder[1];
-      pholder->counts  = 0;
     }
 #endif
   else
     {
       serr("ERROR: Insufficient pre-allocated holders\n");
       pholder          = NULL;
+      DEBUGASSERT(pholder != NULL);
```

Comment:
`DEBUGASSERT(pholder != NULL);` here does not make any sense. It is equivalent to `DEBUGASSERT(0);`.

Reply:
Applied, thanks!